### PR TITLE
[stable/aws-es-proxy] HTTP request signing reverse proxy for Amazon Elasticsearch service

### PR DESCRIPTION
**Pull request by mumoshu**
Manually verified to work by installing it with:

```console
$ helm upgrade esproxy ./ --install \
  --set config.awsSecretAccessKey=MYSECRETKEY \
  --set config.awsAccessKeyID=MYKEYID \
  --set config.endpoint=https://your-aws-es-domain \
  --set config.verbose=true \
  --set podAnnotations.foo=bar
```

From another ubuntu pod, fetching elasticsearch indices on AWS Elasticsearch Service cluster:

```console
root@xenial-1527645050:/# curl esproxy-aws-es-proxy:9200/_cat/indices
yellow open myindex ogrlJsTIQrSJoPvnNMhxxw 5 1 271 0 256.5kb 256.5kb
green  open .kibana    NETaiYeLT-iSu87VSucE9w 1 0   3 1  15.6kb  15.6kb
```

 And then reading logs:

```console
$ stern es-proxy
esproxy-aws-es-proxy-86c8cdc58b-ld6bv aws-es-proxy 2018/05/30 02:14:06 Listening on 0.0.0.0:9200...
esproxy-aws-es-proxy-86c8cdc58b-ld6bv aws-es-proxy 2018/05/30 02:14:09 Generated fresh AWS Credentials object
esproxy-aws-es-proxy-86c8cdc58b-ld6bv aws-es-proxy 2018/05/30 02:14:10  -> GET; 172.17.0.1:54772; /_cluster/health; ; 200; 0.328s
*snip*
esproxy-aws-es-proxy-86c8cdc58b-ld6bv aws-es-proxy 2018/05/30 02:33:36  -> GET; 172.17.0.6:46446; /_cat/indices; ; 200; 0.030s
```

Templates and values.yaml is mostly similar to stable/oauth2-proxy's.
The main difference is `config:` in values.yaml, which directly reflects aws-es-proxy's own configuration file.

### CODE REVIEW

**Comment by mumoshu** [comment]
@foxish Would you mind taking a look at this, or perhaps assign another reviewer if you have any trouble? 🙏 Thanks.

**Comment by mumoshu** [comment]
Could anyone review this?

File: stable/aws-es-proxy/values.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+extraArgs: {}
+  # -log-to-file
+  # -no-sign-reqs
+
+service:
+  type: ClusterIP
+  port: 9200
+  annotations: {}
+  # foo.io/bar: "true"
```
Comment (cpanato) [inline stable/aws-es-proxy/values.yaml]: why this?

**Comment by cpanato** [comment]
@mumoshu this is a new chart, the title should say that.

- please rebase against master and sign the DCO
- The chart version should be at least 1.0.0 since you are pointing that to master
- awsKey and secret are required. Add the `require` in the template to make sure the user is setting that

**Comment by kimxogus** [comment]
@cpanato  I don't think aws key and secret should be required, kube2iam can provide aws credentials in aws k8s clusters.

**Comment by kimxogus** [comment]
@mumoshu I think this chart would be very useful too. Hope this chart merged soon

File: stable/aws-es-proxy/values.yaml (hunk 1 above)
Comment (mumoshu) [reply stable/aws-es-proxy/values.yaml]: As an example to make sure the user gets how to configure annotations. But I'm fine with removing it.

**Comment by mumoshu** [comment]
This should be good to go now.

File: stable/aws-es-proxy/templates/service.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: {{ template "aws-es-proxy.name" . }}
+    chart: {{ template "aws-es-proxy.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+  name: {{ template "aws-es-proxy.fullname" . }}
```
Comment (eviln1) [inline stable/aws-es-proxy/templates/service.yaml:9]: Would you mind making the service name configurable from `values.yaml` ?

Comment (unguiculus) [reply stable/aws-es-proxy/templates/service.yaml:9]: You can use `fullnameOverride` to change it.

**Pull request closed without merging**